src/language-vue/printer-vue.js

```diff
@@ -3,19 +3,33 @@
 const embed = require("./embed");
 const docBuilders = require("../doc").builders;
 const concat = docBuilders.concat;
+const hardline = docBuilders.hardline;
 
 function genericPrint(path, options, print) {
   const n = path.getValue();
   const res = [];
   let index = n.start;
+  let printParent = typeof n.end === "number";
 
   path.each(childPath => {
     const child = childPath.getValue();
     res.push(options.originalText.slice(index, child.start));
     res.push(childPath.call(print));
-    index = child.end;
+    if (typeof child.end === "number") {
+      index = child.end;
+    } else {
+      printParent = false;
+    }
   }, "children");
-  res.push(options.originalText.slice(index, n.end));
+
+  if (printParent) {
+    res.push(options.originalText.slice(index, n.end));
+  }
+
+  // Only force a trailing newline if there were any contents.
+  if (n.tag === "root" && n.children.length) {
+    res.push(hardline);
```

Comment:
What happens if there’s already a trailing (or more) new line, is it going to add one more?

Reply:
It does... I tried to do a right trim on the content but it didn't work.

Reply:
I also tried to normalize this but figured it would not be worth it. I think that the best way forward is to pretty print the first depth of html for vue files. This way we clean up the code a bit more and will handle this newline.

Reply:
Actually no, it works just fine as is.